pkg/pipelines/deployment/deployment.go

```diff
@@ -47,21 +50,21 @@ func ContainerPort(p int32) podSpecFunc {
 }
 
 // Create creates and returns a Deployment with the specified configuration.
-func Create(ns, name, image string, opts ...podSpecFunc) *appsv1.Deployment {
+func Create(partOfLabel, ns, name, image string, opts ...podSpecFunc) *appsv1.Deployment {
 	return &appsv1.Deployment{
 		TypeMeta:   meta.TypeMeta("Deployment", "apps/v1"),
 		ObjectMeta: meta.ObjectMeta(meta.NamespacedName(ns, name)),
 		Spec: appsv1.DeploymentSpec{
 			Replicas: ptr32(1),
-			Selector: labelSelector(KubernetesAppNameLabel, name),
-			Template: podTemplate(name, image, opts...),
+			Selector: LabelSelector(name, partOfLabel),
+			Template: podTemplate(partOfLabel, name, image, opts...),
 		},
 	}
 }
 
 type podSpecFunc func(t *corev1.PodSpec)
 
-func podTemplate(name, image string, opts ...podSpecFunc) corev1.PodTemplateSpec {
+func podTemplate(partOfLabel, name, image string, opts ...podSpecFunc) corev1.PodTemplateSpec {
```

Comment:
You can lose "Label" in partOfLabel and just call it "partOf".

Reply:
made the change